src/sst/core/simulation.cc

```diff
@@ -468,10 +456,41 @@ void Simulation::initialize() {
 
 }
 
-void Simulation::setup() {  
+void Simulation::complete() {
+
+    untimed_phase = 0;
+    // Walk through all the links and call prepareForComplete()
+    for ( auto &i : compInfoMap ) {
+        i->prepareForComplete();
+    }
+
+    syncManager->prepareForComplete();
+
+    bool done = false;
+    completeBarrier.wait();
```

Comment:
Do we need a barrier before the compInfoMap's `prepareForComplete` as well?

Reply:
No, there is a barrier just before complete is called.  Though it could be argued that the barriers between each of the stages in main.cc should be moved into the various calls for the stages.  Any preference here?  Now is a good time to make a change.  Look at main.cc starting at line 319.

Reply:
While I don't think the barrier was necessary, I added it because all the other phases had a barrier right up front.  I think we're doing extra barriers, but they are in a non-time critical section, so I'm not too concerned.  This also means, that if the barriers between phases in main.cc are removed, things should still function correctly.